### PR TITLE
WIP: Add Rails/LeftJoin and Rails/LeftOuterJoin cops

### LINKED ISSUE #5812: Warn using `#joins(‘left outer join …’)` or `#joins(‘lef join …’)` in Rails 5.0 and later
This is either a question or a feature request.

Is it possible to make a cop that warns to use `#left_outer_joins` or `#left_joins` if use `#joins('left outer join ...')` or `#joins('lef join ...')` in Rails 5.0 and later? 

I tried searching but I could not find it, but if you already have such a cop please let me know.

### PR DESCRIPTION
**Pull request by blackst0ne**
This PR adds two cops: `Rails/LeftJoin` and `Rails/LeftOuterJoin` which look for `.joins("LEFT JOIN ...")` and `.joins("LEFT OUTER JOIN ...")` and propose to use the `.left_join("...")` and `.left_outer_join("...")` methods introduced in Rails 5.0.

Closes #5812 

-----------------

Before submitting the PR make sure the following are checked:

* [ ] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [ ] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [ ] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by blackst0ne** [comment]
@bbatsov what's the policy of Rubocop about adding similar cops like in this PR? 

`.joins('LEFT JOIN ...)` -> `.left_join(...)`
`.joins('LEFT OUTER JOIN ...)` -> `.left_outer_join(...)`

Should I create two cops per each method above? 
Or should I just use one cop, say `Rails/LeftJoin`, and check both cases in it? 

:slightly_smiling_face:

**Comment by blackst0ne** [comment]
@Drenmi maybe you could give me a tip (see the previous comment above), please? :slightly_smiling_face:

**Comment by bbatsov** [comment]
@blackst0ne Sorry about the radio silence - I was crazy busy lately. Given the fact that the two checks are pretty similar it seems to me that one cop will be fine. I don't that someone would like to enable one check and disable the other.

**Comment by bbatsov** [comment]
What's the status here? I see the build is till failing.

**Comment by blackst0ne** [comment]
@bbatsov sorry, I was busy last weeks. 
I am going to finish this PR soon. :slightly_smiling_face:

**Comment by blackst0ne** [comment]
@bbatsov specs are green, code is updated. 

Could you review, please? :slightly_smiling_face:

File: CHANGELOG.md [hunk 1]
```diff
@@ -92,6 +92,7 @@
 * [#5406](https://github.com/bbatsov/rubocop/issues/5406): Add new `Layout/ClosingHeredocIndentation` cop. ([@siggymcfried][])
 * [#5823](https://github.com/bbatsov/rubocop/issues/5823): Add new `slashes` style to `Rails/FilePath` since Ruby accepts forward slashes even on Windows. ([@sunny][])
 * New cop `Layout/LeadingBlankLines` checks for empty lines at the beginning of a file. ([@rrosenblum][])
+* [#5871](https://github.com/bbatsov/rubocop/pull/5871): Add new `Rails/LeftJoin` cop. ([@blackst0ne][])
```
Comment (sunny) [inline CHANGELOG.md]: You should bring this up under "master", not under 0.57.0

File: CHANGELOG.md [hunk 2]
```diff
@@ -3491,3 +3492,5 @@
 [@nijikon]: https://github.com/nijikon
 [@mikeyhew]: https://github.com/mikeyhew
 [@kenman345]: https://github.com/kenman345
+[@blackst0ne]: https://github.com/blackst0ne
+
```
Comment (sunny) [inline CHANGELOG.md]: Try and not introduce an empty line here

File: lib/rubocop/cop/rails/left_join.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      class LeftJoin < Cop
+        extend TargetRailsVersion
+
+        minimum_target_rails_version 5.0
+
+        MSG = 'Use `%<prefer>s` instead of `%<current>s`.'.freeze
+
+        PREFERRABLE_QUERIES = {
+          'left join'       => '.left_joins(:model)',
+          'left outer join' => '.left_outer_joins(:model)'
```
Comment (sunny) [inline lib/rubocop/cop/rails/left_join.rb]: Could the `:model` here be inferred by the query to have a better error message?

**Comment by bbatsov** [comment]
This should be moved to rubocop-rails. We won't be accepting any new Rails cops in RuboCop itself anymore.

**Comment by fatkodima** [comment]
@blackst0ne Would you mind reopening this at `rubocop-rails`?

**Pull request closed without merging**